module/VuFind/src/VuFind/Hierarchy/TreeRenderer/HTMLTree.php

```diff
@@ -0,0 +1,265 @@
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  HierarchyTree_Renderer
+ * @author   Luke O'Sullivan <l.osullivan@swansea.ac.uk>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:hierarchy_components Wiki
+ */
+
+namespace VuFind\Hierarchy\TreeRenderer;
+
+use Laminas\Mvc\Controller\Plugin\Url as UrlPlugin;
+use Laminas\View\Renderer\RendererInterface;
+
+use function in_array;
+
+/**
+ * Hierarchy Tree HTML Renderer
+ *
+ * This is a helper class for producing hierarchy trees.
+ *
+ * @category VuFind
+ * @package  HierarchyTree_Renderer
+ * @author   Luke O'Sullivan <l.osullivan@swansea.ac.uk>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:hierarchy_components Wiki
+ */
+class HTMLTree extends AbstractBase implements \VuFind\I18n\Translator\TranslatorAwareInterface
+{
+    use \VuFind\I18n\Translator\TranslatorAwareTrait;
+
+    /**
+     * Router plugin
+     *
+     * @var UrlPlugin
+     */
+    protected $router = null;
+
+    /**
+     * Whether the collections functionality is enabled
+     *
+     * @var bool
+     */
+    protected $collectionsEnabled;
+
+    /**
+     * View renderer
+     *
+     * @var RendererInterface
+     */
+    protected $viewRenderer;
+
+    /**
+     * Constructor
+     *
+     * @param UrlPlugin         $router             Router plugin for urls
+     * @param bool              $collectionsEnabled Whether the collections functionality is enabled
+     * @param RendererInterface $renderer           View renderer
+     */
+    public function __construct(UrlPlugin $router, bool $collectionsEnabled, RendererInterface $renderer)
+    {
+        $this->router = $router;
+        $this->collectionsEnabled = $collectionsEnabled;
+        $this->viewRenderer = $renderer;
+    }
+
+    /**
+     * Get a list of trees containing the item represented by the stored record
+     * driver.
+     *
+     * @param string $hierarchyID Optional filter: specific hierarchy ID to retrieve
+     *
+     * @return mixed An array of hierarchy IDS if an archive tree exists,
```

Comment:
I'm sure this is a copy-pasted legacy issue, but I think we should change the term "archive tree" to "hierarchy tree." That probably comes from some really early code when the terminology was different.

Reply:
Fixed!